src/lib/main.js

```diff
@@ -25,17 +25,26 @@ function chromeMessageHandler(message, sender, sendResponse) {
 }
 
 function parseSettings() {
-	var parsedSettings;
+	var parsedSettings, checkSettings, checkKey, tempSettings;
 
 	if (!localStorage || !localStorage.settings) {
 		parsedSettings = defaultSettings;
-	} else {
+	} else do {
 		try {
-			parsedSettings = JSON.parse(localStorage.settings);
+			checkSettings = JSON.parse(localStorage.settings);
 		} catch (e) {
 			parsedSettings = defaultSettings;
+			break;	// there should really be goto statements in javascript
 		}
-	}
+
+		parsedSettings = {};
+		for (checkKey in defaultSettings)
+			parsedSettings[checkKey] = (checkKey in checkSettings) ? checkSettings[checkKey] : defaultSettings[checkKey];
+
+		tempSettings = JSON.stringify(parsedSettings);
+		if (JSON.stringify(checkSettings) !== tempSettings)
+			localStorage.settings = tempSettings;
+	} while(false);
```

Comment:
All of this object merge logic should go into another function. Then you won't have the urge to write goto statements.